runtime/gc_cheney.c

```diff
@@ -0,0 +1,69 @@
+/**
+ * Here we are implementing the Cheney's algorithm as explained in
+ * https://en.wikipedia.org/wiki/Cheney%27s_algorithm.
+ *
+ */
+
+#include "balrt.h"
+
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <sys/mman.h>
+
+static uint64_t DEFAULT_HEAP_HALF_SIZE = 1073741824; // 1GB
+static uint64_t HEAP_HEADER_SIZE = 8; // in bytes
+
+uint8_t *from_space_ptr;
+uint8_t *to_space_ptr;
+uint8_t *alloc_ptr;
+uint8_t *scan_ptr;
+uint8_t *heap_limit_ptr;
+
+extern void get_roots();
+
+void _bal_init_heap() {
+    int page_size = getpagesize();
+    const char *heap_env = getenv("BAL_HEAP");
+    uint64_t heap_half_size = (heap_env != NULL) ? atoi(heap_env) : DEFAULT_HEAP_HALF_SIZE;
```

Comment:
Need to check return of atoi

Reply:
Fixed in https://github.com/ballerina-platform/nballerina/commit/8964a5ea73124ee8d2132121b66b78e9f0be39a4